app/code/Magento/Braintree/Test/Unit/Gateway/Response/PayPal/VaultDetailsHandlerTest.php

```diff
@@ -142,49 +126,38 @@ public function testHandle()
             'object' => $transaction
         ];
 
-        $this->paymentExtension->expects(static::once())
-            ->method('setVaultPaymentToken')
+        $this->paymentExtension->method('setVaultPaymentToken')
```

Comment:
How does `expects()` make this test more useful? It's better to add additional assert.

Reply:
Currently it is testing that this method **was executed** only **once**, so there were actually 2 checks, that you removed by your changes.

Reply:
I think it's better to add additional `asserts` instead freezing the test code, in that case, it will be more useful than just checking how many times the method was called.

Reply:
@joni-jones looks like you're talking more about integration tests that are testing just that functionality of the class works correct (check outside). 
But unit tests are testing how class behaves in isolation (see what it actually doing), so this actually correct behavior

Reply:
Not only about integration tests. How `expects()` make a tested class isolated? Also, 
there is more than one approach how to write unit tests. https://martinfowler.com/bliki/UnitTest.html